google/api_core/future/polling.py

```diff
@@ -69,57 +100,149 @@ def __init__(self, retry=DEFAULT_RETRY):
         self._done_callbacks = []
 
     @abc.abstractmethod
-    def done(self, retry=DEFAULT_RETRY):
+    def done(self, retry=None):
         """Checks to see if the operation is complete.
 
         Args:
-            retry (google.api_core.retry.Retry): (Optional) How to retry the RPC.
+            retry (google.api_core.retry.Retry): (Optional) How to retry the
+                polling RPC (to not be confused with polling configuration, see
+                the documentation for :meth:`result` for details).
 
         Returns:
             bool: True if the operation is complete, False otherwise.
         """
         # pylint: disable=redundant-returns-doc, missing-raises-doc
         raise NotImplementedError()
 
-    def _done_or_raise(self, retry=DEFAULT_RETRY):
+    def _done_or_raise(self, retry=None):
         """Check if the future is done and raise if it's not."""
-        kwargs = {} if retry is DEFAULT_RETRY else {"retry": retry}
-
-        if not self.done(**kwargs):
+        if not self.done(retry=retry):
             raise _OperationNotComplete()
 
     def running(self):
         """True if the operation is currently running."""
         return not self.done()
 
-    def _blocking_poll(self, timeout=None, retry=DEFAULT_RETRY):
-        """Poll and wait for the Future to be resolved.
+    def _blocking_poll(self, timeout=_DEFAULT_VALUE, retry=None, polling=None):
+        """Poll and wait for the Future to be resolved."""
 
-        Args:
-            timeout (int):
-                How long (in seconds) to wait for the operation to complete.
-                If None, wait indefinitely.
-        """
         if self._result_set:
             return
 
-        retry_ = self._retry.with_deadline(timeout)
+        polling = polling or self._polling
```

Comment:
I wouldn't do it. It's confusing if I don't pass a `polling` or explcitly set it to None, but function will overwrite it. Why don't remove from arg list?

Reply:
Because this the contract all gapic libraries have. Plling or retry configs can be set by users explicitly or, if not set, default ones will be applied. This is a cross-langauge concept. As for why polling was even added in the args in this PR: this is to fix bug in a backward compatible way: originally `retry` argument was viewed as the one controlling polling, although it does not do it, so a new polling argument was added to provide that functionality to the users (i.e. to have an ability to config polling per method call).

Reply:
> so a new polling argument was added to provide that functionality to the users
Users can already configure polling in the constructor. This functionality already exists right ?

Reply:
In which constructor? Please be more specific. Do you mean PollingFuture constructor? Then yes. This is how it works: PollingFuture has a polling config (`self._polling`) which will be  used if user did not specify polling explicitly when they call `future.result()` (i.e. use default one, when "default" is the one passed in the constructor, if the user did not specify their custom config).

Reply:
Okay, this is an anti-pattern then. If you want to configure per method, it shouldn't be in the constructor. However, I see your point that it will be backwards incompatible to remove from constructor, and need to keep per-method config option. No further action needed.